packages/ti_anomali/data_stream/intelligence/elasticsearch/ingest_pipeline/default.yml

```
@@ -0,0 +1,552 @@
+          }
+          if (ctx.threat.indicator == null) {
+            ctx.threat.indicator = [:];
+          }
+          ctx.threat.indicator.type = params[ctx.json.itype];
+        } else {
+          throw new IllegalArgumentException("indicator type not found from itype");
+        }
+      on_failure:
+        - append:
+            field: error.message
+            value: 'Unable to determine STIX 2.0 indicator type from itype "{{{ json.itype }}}": {{{ _ingest.on_failure_message }}}'
+
+  # Detect ipv6 for ipv4-addr types
+  - set:
+      field: threat.indicator.type
+      value: ipv6-addr
+      if: 'ctx.threat?.indicator?.type == "ipv4-addr" && ctx.json.ip != null && ctx.json.ip.contains(":")'
+
+  # Map IP geolocation fields.
+  - convert:
+      field: json.latitude
+      target_field: threat.indicator.geo.location.lat
+      type: double
+      if: "ctx.json?.latitude != null && ctx.json?.longitude != null"
+      on_failure:
+        - append:
+            field: error.message
+            value: 'Cannot convert latitude field "{{{ json.latitude }}}" to double: {{{ _ingest.on_failure_message }}}'
+  - convert:
+      field: json.longitude
+      target_field: threat.indicator.geo.location.lon
+      type: double
+      if: "ctx.json?.latitude != null && ctx.json?.longitude != null"
+      on_failure:
+        - append:
+            field: error.message
+            value: 'Cannot convert longitude field "{{{ json.longitude }}}" to double: {{{ _ingest.on_failure_message }}}'
+  - remove:
+      field:
+        - json.latitude
+        - json.longitude
+      ignore_missing: true
+
+  - rename:
+      field: json.tlp
+      target_field: threat.indicator.marking.tlp
+      ignore_missing: true
+  - uppercase:
+      field: threat.indicator.marking.tlp
+      ignore_missing: true
+  - set:
+      field: threat.indicator.marking.tlp
+      value: AMBER
+      if: 'ctx.threat?.indicator?.marking?.tlp == null && !ctx.json.is_public'
+  - set:
+      field: threat.indicator.marking.tlp
+      value: WHITE
+      if: 'ctx.threat?.indicator?.marking?.tlp == null && ctx.json.is_public'
+
+  - script:
+      lang: painless
+      if: ctx.json?.confidence != null
+      description: Normalize confidence to STIX 2.1 values
+      source: >
+        if (ctx.threat == null) {
+          ctx.threat = [:];
+        }
+        if (ctx.threat.indicator == null) {
+          ctx.threat.indicator = [:];
+        }
+        def value = ctx.json.confidence;
+        if (value == null) {
+          ctx.threat.indicator.confidence = "Not Specified";
+        } else if (value <= 0.0 || value > 100.0) {
+          ctx.threat.indicator.confidence = "None";
+        } else if (value >= 1.0 && value <= 29.0) {
+          ctx.threat.indicator.confidence = "Low";
+        } else if (value >= 30.0 && value < 70) {
+          ctx.threat.indicator.confidence = "Medium";
+        } else {
+          ctx.threat.indicator.confidence = "High";
```

```suggestion
        } else if (value <= 0.0 || 100.0 < value) {
          ctx.threat.indicator.confidence = "None";
        } else if (value < 30.0) {
          ctx.threat.indicator.confidence = "Low";
        } else if (value < 70.0) {
          ctx.threat.indicator.confidence = "Medium";
        } else {
          ctx.threat.indicator.confidence = "High";
```
Currently this will give a confidence of "High" for 0 < values < 1 or 29 < values < 30.

Reordering of first guard to align with number line (read, "value is outside the range [0,100)").